### DISCUSSION TITLE
Specify length of trajectory

### DISCUSSION THREAD

**Discussion by jblich870** [GreenleafLab/ArchR#1729] [Feature Requests]
Hi,

Would it be possible to add a parameter to addTrajectory that allows one to set the length of a trajectory?

One option might be to specify the number of knots as in `stats::smooth.spline()`(https://rdrr.io/r/stats/smooth.spline.html)

My trajectory shows a weird kink at the end regardless of which parameters I use within addTrajectory, that ideally I would cut out:
![image](https://user-images.githubusercontent.com/69856293/200553776-f65b43e8-6dec-420d-90e8-b93e468f38fe.png)

Many thanks,
J

**Comment by jblich870** [comment]
Just to clarify, does the shape of the arrow generated by `plotTrajectory` represent the actual continuous trajectory as described in the manual: "ArchR fits a continuous trajectory to each N-dimensional coordinate based on the pseudo-time value using the smooth.spline function"?

I assume this is then used to generate pseudotime values for each cell. If so, the fact that it bends at the end is problematic for my analysis as gives weird ordering of cells at the end.

**Reply by rcorces** [reply]
Sorry for the delay in replying. A few clarifying points to address your questions here and in your other post (https://github.com/GreenleafLab/ArchR/discussions/1689#discussioncomment-4138356).

The spline used to calculate the trajectory and the arrow displayed on the trajectory are not directly related. The spline is created and used to determine pseudotime of the trajectory. The arrow, on the other hand, is generated for visualization purposes during `plotTrajectory()` and does not utilize the spline directly but rather the pseudotime values. To be clear, the arrow is not used to generate pseudotime.

https://github.com/GreenleafLab/ArchR/blob/154d9a1837a0947270adc2a483d4feeebeb3dffe/R/Trajectory.R#L1027-L1041

So, in some ways, the arrow is arbitrary and meaningless and is just a representation of pseudotime. I think it is fine to manually manipulate the arrow in post as long as you make it clear that the arrow is merely a representation and was manually drawn.

**Reply by jblich870** [reply]
Hi,

Apologies for the delayed reply - many thanks for clarifying, that's really helpful.

Best wishes,
J
